python/pyspark/pandas/tests/test_series.py

```diff
@@ -1333,6 +1333,17 @@ def test_rank(self):
         self.assert_eq(pser.rank(method="first"), psser.rank(method="first").sort_index())
         self.assert_eq(pser.rank(method="dense"), psser.rank(method="dense").sort_index())
 
+        non_numeric_pser = pd.Series(["a", "c", "b", "d"], name="x", index=[10, 11, 12, 13])
+        non_numeric_psser = ps.from_pandas(non_numeric_pser)
+        self.assert_eq(
+            non_numeric_pser.rank(numeric_only=True).sort_index(),
+            non_numeric_psser.rank(numeric_only=True).sort_index(),
+        )
+        self.assert_eq(
+            (non_numeric_pser + "x").rank(numeric_only=True).sort_index(),
+            (non_numeric_psser + "x").rank(numeric_only=True).sort_index(),
+        )
```

Comment:
```suggestion
        self.assert_eq(
            non_numeric_pser.rank(numeric_only=True),
            non_numeric_psser.rank(numeric_only=True),
        )
        self.assert_eq(
            (non_numeric_pser + "x").rank(numeric_only=True),
            (non_numeric_psser + "x").rank(numeric_only=True),
        )
```

I guess `sort.index` is useless in here, because we will get a empty series